docs/sphinx/developer/components/branchPred.rst

```diff
@@ -3,28 +3,35 @@ Branch prediction
 
-The algorithm(s) held within a ``BranchPredictor`` class instance can be model-specific, however, SimEng provides a set of predictors detailed below.
+The ``predict`` function is passed an instruction address, branch type, and a possible known target. The branch type argument currently supports the following types:
 
-AlwaysNotTakenPredictor
------------------------
+- Conditional
+- LoopClosing
+- Return
+- SubroutineCall
+- Unconditional
 
-The ``AlwaysNotTakenPredictor`` is the simplest available branch predictor. It is a static predictor, only providing a "not taken" branch direction with no branch target prediction and no update functionality.
+The usage of these parameters within a branch predictor's ``predict`` function is algorithm specific.
 
+The ``update`` function is passed the branch outcome, the instruction address, and the branch type. From this information, any algorithms or branch structures may be updated.
 
-BTBPredictor
-------------
+Generic Predictor
+-----------------
 
-A more complex branch predictor is the ``BTBPredictor`` utilising a Branch Target Buffer (BTB) and a branch direction buffer. Both structures are indexed by the instruction address, after the application of a mask, and store previous outcomes of executed branches.
+The algorithm(s) held within a ``BranchPredictor`` class instance can be model-specific, however, SimEng provides a ``GenericPredictor`` which contains the following logic.
 
-Access to a structure entry with no previous history will supply a "not taken" direction prediction with no known target, similar to the ``AlwaysNotTakenPredictor``.
+Global History
+    For indexing relevant prediction structures, a global history can be utilised. The global history value uses n-bits to store the n most recent branch direction outcomes, with the left-most bit being the oldest.
 
-BTB_BWTPredictor
-----------------
+Branch Target Buffer (BTB)
+    For each entry, the BTB stores the most recent target along with an n-bit saturating counter for an associated direction. The indexing of this structure uses the lower bits of an instruction address XOR'ed with the current global branch history value.
 
-Based on the algorithms used by the A64FX processor, the ``BTB_BWTPredictor`` branch predictor utilises a 4-way associative BTB-style structure with each entry storing a 2-bit direction agreement policy and branch target. An agreement policy assigns an agreement bias value against a chosen branch direction such that, ``11`` relates to a strong agreement and ``00`` a strong disagreement.
+    If the supplied branch type is Unconditional, then the predicted direction is overridden to be taken. If the supplied branch type is Conditional and the predicted direction is not taken, then the predicted target is overridden to be the next sequential instruction.
```

Comment:
"Unconditional" capitalisation? If you are referring to the parameter value should it be in quotes (or some consistent punctuation)

Reply:
Same for "Conditional"